src/daily-quest/daily-quest.routes.ts

```diff
@@ -1,20 +1,48 @@
-import { Router } from "express";
+import { RequestHandler, Router } from "express";
 import DailyQuestController from "./daily-quest.controller";
 import { queryParamsParser } from "@/middleware/queryParams";
+import { auth } from "@/middleware/authentication";
+import { routeParamsParser } from "@/middleware/routeParams";
 
 const dailyQuestRouter = Router();
 const dailyQuestController = new DailyQuestController();
 
-dailyQuestRouter.get("/", queryParamsParser, dailyQuestController.getAll);
-dailyQuestRouter.get("/:id", dailyQuestController.getById);
-dailyQuestRouter.post("/", dailyQuestController.create);
+dailyQuestRouter.get(
+  "/",
+  auth,
+  queryParamsParser,
+  dailyQuestController.getAll as RequestHandler
+);
 dailyQuestRouter.get(
   "/for-date",
+  auth,
   queryParamsParser,
-  dailyQuestController.getForDate
+  dailyQuestController.getForDate as RequestHandler
+);
+dailyQuestRouter.get(
+  "/:id",
+  auth,
+  routeParamsParser,
+  dailyQuestController.getById as RequestHandler
+);
+dailyQuestRouter.post("/", auth, dailyQuestController.create as RequestHandler);
+dailyQuestRouter.patch(
+  "/:id",
+  auth,
+  routeParamsParser,
+  dailyQuestController.update as RequestHandler
+);
+dailyQuestRouter.delete(
+  "/:id",
+  auth,
+  routeParamsParser,
+  dailyQuestController.delete as RequestHandler
```

Comment:
can these types casting be performed at controller level, so you do not need to do that here?

Reply:
Unfortunately, no. At least, I didn't find the way((

This is needed to support user in Authenticated request:
![image](https://github.com/user-attachments/assets/4a4b0933-8b7c-4bbc-a045-cbe4def1cd87)


I'd appreciate any help with this as don't like this ugly workaround as well.

Reply:
The best solution I could come to is to wrap controller call with a wrapper:
![image](https://github.com/user-attachments/assets/24340ce5-fe45-4218-b068-b5451b31f24c)


![image](https://github.com/user-attachments/assets/569fd6f6-3d10-4fb4-8eb9-fd8dff10ce3a)